sync/src/block/downloader/header.rs

```diff
@@ -74,9 +74,16 @@ impl HeaderDownloader {
     }
 
     pub fn update(&mut self, total_score: U256, best_hash: H256) -> bool {
-        if self.total_score < total_score {
+        if self.total_score <= total_score {
             self.total_score = total_score;
             self.best_hash = best_hash;
+
+            if self.client.block_header(&BlockId::Hash(best_hash)).is_some() {
```

Comment:
I suspect that this statement caused the deadlock.

Reply:
I found the below log when applied this commit to Corgi.
"cannot timeout 0: "SendError"

Reply:
AFAIK, `SendError` is occurred only when the receiver of the channel had been closed because we used an unbounded channel. Please check whether the extension thread had been crashed.

Reply:
Thanks to seulgi, I found that the error was caused by a log statement which is deleted in this PR.